views.py

```diff
@@ -100,7 +100,7 @@ def rights_edit(request, pid, dsid):
     if form.is_valid():
         new_rights = form.build_rights()
         params = {'pid': pid}
-        params['rights'] = json.dumps({'xml_data': new_rights})
+        params['rights'] = json.dumps({'xml_data': new_rights.serializeDocument()})
```

Comment:
I think current objects are being created with serialize() content, instead of serializeDocument().